### PR TITLE
[Improvement] Optimize paimon to store monitoring data, causing threads not to be released

### LINKED ISSUE #3343: [Improvement] CPU is too high
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar issues.

### What happened

If I don't log in to Dinky over the weekend, my CPU will soar to 80% when I log in on Monday, and I can't log in anymore. The only solution is to restart. I'm not sure if it's caused by Dinky's built-in monitoring.

### What you expected to happen

If I don't log in to Dinky over the weekend, my CPU will soar to 80% when I log in on Monday, and I can't log in anymore. The only solution is to restart. I'm not sure if it's caused by Dinky's built-in monitoring.

### How to reproduce

If I don't log in to Dinky over the weekend, my CPU will soar to 80% when I log in on Monday, and I can't log in anymore. The only solution is to restart. I'm not sure if it's caused by Dinky's built-in monitoring.

### Anything else

_No response_

### Version

1.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Pandas886**
## Purpose of the pull request

close issue #3343
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by Pandas886** [comment]
使用固定线程池和提高攒批量写入间隔

File: dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java [hunk 1]
```diff
@@ -40,37 +40,47 @@
 public class MetricsContextHolder {
 
     protected static final MetricsContextHolder instance = new MetricsContextHolder();
+    private final List<MetricsVO> metricsVOS = new CopyOnWriteArrayList<>();
+    private final AtomicLong lastDumpTime = new AtomicLong(System.currentTimeMillis());
 
     public static MetricsContextHolder getInstances() {
         return instance;
     }
+    // 创建具有自定义命名的ThreadFactory
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java]: Use English annotations

File: dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java [hunk 2]
```diff
@@ -40,37 +40,47 @@
...
+            new ThreadFactoryBuilder().setNameFormat("metrics-send-thread-%d").build();
 
-    /**
-     * Temporary cache monitoring information, mainly to prevent excessive buffering of write IO,
-     * when metricsVOS data reaches 1000 or the time exceeds 5 seconds
-     */
-    private final List<MetricsVO> metricsVOS = Collections.synchronizedList(new ArrayList<>());
-
-    private final Long lastDumpTime = System.currentTimeMillis();
+    // 创建自定义的ThreadPoolExecutor
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java]: Use English annotations

File: dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java [hunk 3]
```diff
@@ -40,37 +40,47 @@
...
+                List<MetricsVO> snapshot;
+                synchronized (this) { // 仅在需要操作时才进入同步块
+                    snapshot = new ArrayList<>(metricsVOS);
+                    metricsVOS.clear();
+                    lastDumpTime.set(current);
+                }
+                PaimonUtil.write(PaimonTableConstant.DINKY_METRICS, snapshot, MetricsVO.class);
+            }
+            String topic = StrFormatter.format("{}/{}", SseTopic.METRICS.getValue(), key);
+            SseSessionContextHolder.sendTopic(topic, o); // 确保只有成功添加的指标才发送
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/context/MetricsContextHolder.java]: Use English annotations

**Comment by Zzm0809** [comment]
需要执行下代码格式化

**Review by Zzm0809 — APPROVED** [review]

**Pull request merged**